src/ssh/sftp_dir_iterator.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2019-2022 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <multipass/ssh/sftp_dir_iterator.h>
+
+#include <multipass/ssh/sftp_utils.h>
+
+#include <multipass/format.h>
+#include <stdexcept>
+
+namespace multipass
+{
+void SFTPDirIterator::push_dir(const fs::path& path)
+{
+    auto dir = mp_sftp_opendir(sftp, path.u8string().c_str());
+    if (!dir)
+        throw std::runtime_error{
+            fmt::format("[sftp] cannot open remote directory {}: {}", path, ssh_get_error(sftp->session))};
```

Comment:
I think a blank line here would help a little with readability.

Reply:
where exactly? after line 32 or 29?

Reply:
Ugh, stupid Github comments don't make it clear where exactly the reviewer was referring to.  I'm talking about adding a blank line between line 32 and line 33.